microcloud/service/microcloud.go

```diff
@@ -127,19 +127,28 @@ func (s CloudService) RequestJoin(ctx context.Context, secret string, name strin
 	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
 	defer cancel()
 
-	c, err := s.client.RemoteClient(util.CanonicalNetworkAddress(joinConfig.Address, CloudPort))
-	if err != nil {
-		return err
-	}
+	var c *microClient.Client
+	var err error
+	if name == s.name {
+		c, err = s.client.LocalClient()
+		if err != nil {
+			return err
+		}
+	} else {
+		c, err = s.client.RemoteClient(util.CanonicalNetworkAddress(joinConfig.Address, CloudPort))
+		if err != nil {
+			return err
+		}
 
-	c.Client.Client.Transport = &http.Transport{
-		TLSClientConfig:   &tls.Config{InsecureSkipVerify: true},
-		DisableKeepAlives: true,
-		Proxy: func(r *http.Request) (*url.URL, error) {
-			r.Header.Set("X-MicroCloud-Auth", secret)
+		c.Client.Client.Transport = &http.Transport{
+			TLSClientConfig:   &tls.Config{InsecureSkipVerify: true},
+			DisableKeepAlives: true,
+			Proxy: func(r *http.Request) (*url.URL, error) {
+				r.Header.Set("X-MicroCloud-Auth", secret)
 
-			return shared.ProxyFromEnvironment(r)
-		},
+				return shared.ProxyFromEnvironment(r)
+			},
+		}
```

Comment:
Now that https://github.com/canonical/microcluster/pull/83 is merged does this need to be updated? Same for the other cases where this is being set.

Reply:
Instead of filling this PR with all of those changes, I've added a new PR that moves all this implementation into a helper so we can just call that instead: #287

Reply:
Ok so https://github.com/canonical/microcloud/pull/287 has to come after this one then?